ecosystem/indexer-grpc/indexer-grpc-file-store/src/processor.rs

```diff
@@ -87,109 +84,148 @@ impl Processor {
+    ///   3.2 If we're ready to process, create max of 10 threads and fetch / upload data
+    ///   3.3 Update file store metadata at the end of a batch
     pub async fn run(&mut self) -> Result<()> {
         let cache_chain_id = self.cache_chain_id;
 
-        // If file store and cache chain id don't match, return an error.
-        let metadata = self
-            .file_store_operator
-            .create_default_file_store_metadata_if_absent(cache_chain_id)
-            .await
-            .context("Metadata did not match.")?;
+        let mut batch_start_version =
+            if let Some(metadata) = self.file_store_operator.get_file_store_metadata().await {
+                anyhow::ensure!(metadata.chain_id == cache_chain_id, "Chain ID mismatch.");
+                metadata.version
+            } else {
+                0
+            };
 
-        // This implements a two-cursor approach:
-        //   * One curosr is to track the current cache version.
-        //   * The other cursor is to track the current file store version.
-        //   * Constrains:
-        //     * The current cache version >= the current file store version.
-        //     * The current file store version is always a multiple of BLOB_STORAGE_SIZE.
-        let mut current_cache_version = metadata.version;
-        let mut current_file_store_version = current_cache_version;
-        // The transactions buffer to store the transactions fetched from cache.
-        let mut transactions_buffer: Vec<EncodedTransactionWithVersion> = vec![];
         let mut tps_calculator = MovingAverage::new(10_000);
         loop {
-            // 0. Data verfiication.
-            // File store version has to be a multiple of BLOB_STORAGE_SIZE.
-            if current_file_store_version % BLOB_STORAGE_SIZE as u64 != 0 {
-                bail!("File store version is not a multiple of BLOB_STORAGE_SIZE.");
+            let latest_time = std::time::Instant::now();
+            let cache_worker_latest = self.cache_operator.get_latest_version().await?;
+
+            // batches tracks the start version of the batches to fetch. 1000 at the time
+            let mut batches = vec![];
+            let mut start_version = batch_start_version;
+            while start_version + (BLOB_STORAGE_SIZE as u64) < cache_worker_latest {
+                batches.push(start_version);
+                start_version += BLOB_STORAGE_SIZE as u64;
             }
 
-            let file_store_upload_batch_start = std::time::Instant::now();
-            let batch_get_result = self
-                .cache_operator
-                .batch_get_encoded_proto_data(current_cache_version)
+            // we're too close to the head
+            if batches.is_empty() {
+                debug!(
+                    batch_start_version = batch_start_version,
+                    cache_worker_latest = cache_worker_latest,
+                    "[Filestore] No enough version yet, need 1000 versions at least"
```

Comment:
can you include service_type here

Reply:
this is a debug log. it should never get read. I don't think that it has to follow the same format as the info logs.